test/performance/odp_queue_perf.c

```diff
@@ -167,29 +184,36 @@ static int parse_options(int argc, char *argv[], test_options_t *test_options)
 		}
 	}
 
-	if (test_options->num_queue > MAX_QUEUES) {
-		ODPH_ERR("Too many queues %u. Test maximum %u.\n",
+	if (test_options->num_queue > MAX_QUEUES || test_options->num_queue == 0) {
+		ODPH_ERR("Invalid number of queues %u. Test maximum %u.\n",
 			 test_options->num_queue, MAX_QUEUES);
 		return -1;
 	}
 
-	if (test_options->single && !test_options->private_queues && test_options->num_cpu != 1) {
-		ODPH_ERR("Using single prod/cons queue(s) with multiple workers not supported.\n");
-		return -1;
-	}
+	num_cpu = test_options->num_cpu;
+	if (num_cpu == 0)
+		num_cpu = odp_cpumask_default_worker(NULL, 0);
 
 	if (test_options->private_queues) {
-		int num_cpu = test_options->num_cpu;
-
-		if (num_cpu == 0)
-			num_cpu = odp_cpumask_default_worker(NULL, 0);
-
 		if ((int)test_options->num_queue < num_cpu) {
 			ODPH_ERR("Not enough queues for %d workers.\n", num_cpu);
 			return -1;
 		}
 	}
 
+	if (test_options->single && !test_options->private_queues) {
+		if ((test_options->mode == TEST_MODE_LOOP && num_cpu != 1) ||
+		    (test_options->mode == TEST_MODE_PAIR && num_cpu != 2)) {
+			ODPH_ERR("Using single prod/cons queue(s) with multiple workers not allowed.\n");
```

Comment:
I guess the text is not true anymore. Could be rephrased to mention multiple producers/consumers not allowed with single prod/cons queues.

Reply:
Updated in V3.